include/swift/AST/DiagnosticsSIL.def

```diff
@@ -350,6 +350,13 @@ NOTE(constexpr_called_from,none, "when called from here", ())
 NOTE(constexpr_not_evaluable,none,
      "expression not evaluable as constant here", ())
 
+NOTE(constexpr_unknown_control_flow_due_to_skip,none,
+    "branch depends on non-constant value obtained by skipping instructions",())
+NOTE(constexpr_returned_by_skip,none, "return value of a skipped instruction "
+     "is not a constant", ())
+NOTE(constexpr_mutated_by_skip,none, "value mutatable by a skipped instruction "
+    "is not a constant", ())
+
```

Comment:
"skipped instruction" doesn't seem like a concept that should be exposed to the user.

In context of the os_log client, users might understand something like "result of an unrecognized operation" / "mutated by an unrecognized operation". What do you think about changing the diagnostic messages to those?

Reply:
That right. It is not a user concept. It added this note for use in tests. It is not a part of the os_log client.  Ideally, I would like to move all diagnostics to clients (like #assert client, tester client etc.) and specialize them as needed.  In some sense, they are a part of the client model on how they handle errors in evaluation, as the interpreter itself is not a user-level concept, as of now.